src/admin/dto/update-admin.dto.ts

```diff
@@ -0,0 +1,25 @@
+import { IsEmail, IsNotEmpty, IsString } from "class-validator";
+
+export class UpdateAdminDto {
+  @IsNotEmpty()
+  @IsEmail()
+  readonly email: string;
+
+  readonly password: string;
+
+  readonly hashedRefreshToken: string;
+
+  readonly createdAt: Date;
+
+  readonly updatedAt: Date;
+
+  @IsNotEmpty()
+  @IsString()
+  readonly permission: string;
+
+  readonly isDeleted: boolean;
+
+  @IsNotEmpty()
+  @IsString()
+  readonly name: string;
+}
```

Comment:
A small suggestion. I saw the update admin dto is almost a copy of create admin dto. I believe you can try to use partial type.

`export class UpdateAdminDto extends PartialType(CreateAdminDto) {
  xxxxx // your changes
}`

Reply:
thx modified it to PartialType

`import { PartialType } from "@nestjs/mapped-types";
import { CreateAdminDto } from "./create-admin.dto";

export class UpdateAdminDto extends PartialType(CreateAdminDto) {}
`